data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -545,6 +546,37 @@ private StateSyncCommitteesData getSyncCommitteesFromState(
             committeeIndices, spec.atEpoch(epoch).getConfig().getTargetCommitteeSize()));
   }
 
+  @VisibleForTesting
+  protected Map<Integer, Integer> getCommitteeIndices(
+      final List<BLSPublicKey> committeeKeys,
+      final Set<String> validators,
+      final tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState state) {
+    if (validators.isEmpty()) {
+      final List<Integer> result =
+          committeeKeys.stream()
+              .flatMap(pubkey -> spec.getValidatorIndex(state, pubkey).stream())
+              .collect(toList());
+
+      return IntStream.range(0, result.size())
+          .boxed()
+          .collect(Collectors.<Integer, Integer, Integer>toMap(Function.identity(), result::get));
+    }
+
+    final Map<Integer, Integer> output = new HashMap<>();
+    int i = 0;
+    for (BLSPublicKey key : committeeKeys) {
```

Comment:
i think this would be cleaner if you make a for loop in a more traditional way
```
for (int i = 0; i < comitteeKeys.length(); i++) {
```
and instead of key use `committeeKeys.get(i)`
or even do final `BlsPublicKey key = committeeKeys.get(i);` as first line potentially